qiskit_nature/list_or_dict.py

```diff
@@ -34,6 +35,15 @@ def __init__(self, values: Optional[ListOrDictType] = None):
         Args:
             values: an optional object of `list` or `dict` type.
         """
+        warn_deprecated(
+            "0.5.0",
+            old_type=DeprecatedType.CLASS,
+            old_name="qiskit_nature.ListOrDict",
+            additional_msg=(
+                ". Qiskit Nature will now always produce dictionary-style auxiliary operators."
+            ),
```

Comment:
How does this relate to the now deprecated code. Presumably that still uses the settings flag and will produce a list or dict based on the settings. By default it will now produce a dict, but presumably a user could change the settings back to have lists. Its ignored in the new code of course so the above message seems to be true for that. But then the type is not used either right.

Reply:
The new code ignores this setting now.
The deprecated code now also produces dict by default, but the user can change it back to list.